src/apps/answers/service.py

```diff
@@ -568,6 +580,40 @@ async def get_activity_answer(
         )
 
         return submission
+    
+    async def _validate_relation_between_subjects_in_applet(
+            self,
+            respondent_subject: SubjectSchema,
+            target_subject: SubjectSchema,
+            source_subject: SubjectSchema
+    ) -> None:
+        
+        """
+            add a new relation check in here, 
+            1. check if there is an actual relation (logged-in user (respondent_subject) and targetSubject)
+            2. another relation between respondent_subject and the subject it self
+            - respondent is the one who is inputting the answers in the activity
+            - targetSubject: the one the responses about
+            - sourceSubject: who will be providing the responses
+        """
```

Comment:
Comment is a good start. How about this for further clarity

```suggestion
    """
    Validate the relationship between subjects in an applet.
    
    - respondent_subject: the subject inputting the answers.
    - target_subject: the subject about whom the responses are.
    - source_subject: the subject providing the responses.
    
    This method checks:
    1. If there is a relationship between the logged-in user (respondent_subject) and target_subject.
    2. If there is a relationship between the respondent_subject and the source_subject.

    Raises:
        MultiinformantAssessmentNoAccessApplet: If no valid relationship is found.
    """
```

Reply:
thank you so much Marty
I am going to remove these comments I left them just as a reminder of the functionality so I do not get lost but your comment is way better and will help me understand more so thank you for the changes! 🙌🏻